src/views/Results.vue

```diff
@@ -0,0 +1,31 @@
+<template>
+  <section>
+    <button @click="selectView">List View</button>
+    <button @click="selectView">Map View</button>
+    <results-list v-if="listView"></results-list>
+    <results-map v-else></results-map>
+  </section>
+</template>
+
+<script>
+import ResultsList from '../components/ResultsList.vue'
+import ResultsMap from '../components/ResultsMap.vue'
+export default {
+  data() {
+    return {
+      listView: true,
+    };
+  },
+  components: {
+    ResultsList,
+    ResultsMap
+  },
+  methods: {
+    selectView() {
+      this.listView = !this.listView;
+    },
+  },
+};
+</script>
```

Comment:
Okay, so I just want to confirm what is happening here.  If the user clicks on the `List View` or `Map View` button, the `selectView()` method is invoked.  You chose to have one piece of data, `listView`, set to true by default.  Each time the user clicks either button, this method will set the opposite value of true or false based on its current state.  This will then trigger a re-render similar to React, where `listView`'s value has changed and that determines if the map is shown (when set to false).  Anything I should change or add to my understanding?

Reply:
Agreed with Nicole on this one. I pulled down the branch to test this out. The `listView` is set to `true` by default, I believe this is intended for a user to see the `results-list` component when they first come to this page. If a user clicks on a button for a page they are already on, let's say `List View` as an example, it will still trigger the `selectView()` method and change it from true to false, even though they are currently on the `List View` component and not the `results-map`. This sounds like it would not quite operate the way it is expecting and have an unintended result.

Reply:
Ah, great point. Looks like we have our first bug! I'm thinking we should add an if/else to `selectView()`? Any other ideas?

Reply:
I think an `if/else` would work really well here. Vue still allows us to pass arguments/parameters even though it looks a little strange seeing it directly inside the HTML, or custom component. Giving them a specific button they clicked on as an argument should allow us to detect which one they click and change appropriately.

Reply:
I just learned about the `<component>` tag specific to Vue that I think would work perfectly here. I'll tell y'all about it tomorrow.

Reply:
That's a really good point! Completely forgot about the `<component>` tag. With a `:is` bind declarative, that could work really well with the key/value pair you have setup in `data()`.

Reply:
Excited to learn more about this, Stacy! Thanks for that catch, Nathan.  We could set a conditional, or explore the suggestion from Stacy.  Another idea would be to set up a second method that handles switching `listView` to false, and allowing `selectView()` to only change it to true rather than its inverse value.  Just another possibility, although I'm thinking I like the two other ideas better!